tango/integrations/beaker/common.py

```diff
@@ -95,6 +172,7 @@ def release(self):
                 # Dataset must have been manually deleted.
                 pass
             self._lock_dataset = None
+            atexit.unregister(self.release)
```

Comment:
`atexit` works properly with methods?

Reply:
Apparently! Although one side-effect is that if you have an acquired lock and call `del lock`, the `__del__` method won't actually be called immediately since a reference to the lock will still exist in the `atexit` hook. That's not ideal, and wouldn't be okay IMO if `BeakerStepLock` was meant to be used outside of Tango. But since it's only used internally I think that's okay. Just means we have to explicitly call `lock.release()` instead of relying on `lock.__del__()`.